src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -258,6 +261,65 @@ private static boolean isPresent(String availableChecks, String checkName) {
         return availableChecks.matches(linkPattern);
     }
 
+    private static Map<String, String> readSummaries() throws Exception {
+        final String fileName = AVAILABLE_CHECKS_PATH.getFileName().toString();
+        final String input = new String(Files.readAllBytes(AVAILABLE_CHECKS_PATH), UTF_8);
+        final Document document = XmlUtil.getRawXml(fileName, input, input);
+        final NodeList rows = document.getElementsByTagName("tr");
+        final Map<String, String> result = new HashMap<>();
+
+        for (int position = 0; position < rows.getLength(); position++) {
+            final Node row = rows.item(position);
+            final Iterator<Node> cells = XmlUtil.findChildElementsByTag(row, "td").iterator();
+            final String name = XmlUtil.sanitizeXml(cells.next().getTextContent());
+            final String summary = XmlUtil.sanitizeXml(cells.next().getTextContent());
+
+            result.put(name, summary);
+        }
+
+        return result;
+    }
+
+    @Test
+    public void testAllChecksPageInSyncWithChecksSummaries() throws Exception {
+        final Pattern endOfSentence = Pattern.compile("(.*?\\.)\\s", Pattern.DOTALL);
+        final Map<String, String> summaries = readSummaries();
+
+        for (Path path : XdocUtil.getXdocsConfigFilePaths(XdocUtil.getXdocsFilePaths())) {
+            final String fileName = path.getFileName().toString();
+            if ("config_reporting.xml".equals(fileName)
+                    || "config_filefilters.xml".equals(fileName)
+                    || "config_filters.xml".equals(fileName)) {
```

Comment:
@romani Is there any reason filters don't have their own summary page?

Reply:
No good reason, but they already all at one page https://checkstyle.org/config_filters.html and one more page for file filters